common/hooks/useSavedSearchState.ts

```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect, Dispatch, SetStateAction } from 'react';
+import { WorksProps } from '../views/components/WorksLink/WorksLink';
+
+let savedState;
```

Comment:
I see why/how this works but it does seem _weird_ and I imagine changes in SSR could break it unexpectedly. Not sure what else to suggest though, perhaps something could be done using context APIs?

Separately I'm not sure about "saved" - I guessed that it was going to use localstorage, maybe `usePreviousSearchState`? Again not sure on a better option.

Reply:
Reasonably confident that context is the "right" way to do it without mutation and closure etc etc, but if you want to get this PR in first and make that change next (I am also happy to have a go at it) that makes sense

Reply:
Na, we had it like that before. I'll change it back.